src/process/process_set.cu

```diff
@@ -214,15 +209,43 @@ namespace micm
       return devstruct;
     }
 
-    /// This is the function that will copy the jacobian_flat_id
-    ///   of class "ProcessSet" to the device, after the matrix
-    ///   structure is known;
-    void CopyJacobiFlatId(ProcessSetParam& hoststruct, ProcessSetParam& devstruct)
+    /// This is the function that will copy the information needed
+    /// to calculate Jacobian terms to the device
+    void CopyJacobianParams(ProcessSetParam& hoststruct, ProcessSetParam& devstruct)
     {
       /// Calculate the memory space
+      size_t jacobian_process_info_bytes = sizeof(ProcessInfoParam) * 
+                                             hoststruct.jacobian_process_info_size_;
+      size_t jacobian_reactant_ids_bytes = sizeof(size_t) * hoststruct.jacobian_reactant_ids_size_;
+      size_t jacobian_product_ids_bytes = sizeof(size_t) * hoststruct.jacobian_product_ids_size_;
+      size_t jacobian_yields_bytes = sizeof(double) * hoststruct.jacobian_yields_size_;
       size_t jacobian_flat_ids_bytes = sizeof(size_t) * hoststruct.jacobian_flat_ids_size_;
 
       /// Allocate memory space on the device
+      CHECK_CUDA_ERROR(
+          cudaMallocAsync(
+              &(devstruct.jacobian_process_info_),
+              jacobian_process_info_bytes,
+              micm::cuda::CudaStreamSingleton::GetInstance().GetCudaStream(0)),
```

Comment:
can we save `micm::cuda::CudaStreamSingleton::GetInstance().GetCudaStream(0)` to a local variable and simplify the interface here?